File: PreNationals/Contests/AtCoder/abc190_d.py
from math import sqrt
n = int(input())
n = 2*n
ans = 0
upTo = sqrt(n)
for i in range(1,int(upTo+1)):
    if(n % i == 0):
        check = n - (i*i)+i
        if(check % (2*i) == 0): ans += 1
        ot = n // i
        if(ot != i):
            check = n - (ot*ot)+ot
            if(check % (2*ot) == 0): ans += 1

print(ans)

File: PreNationals/Contests/AtCoder/abc190_d.cpp
#include<bits/stdc++.h>
#define ll long long
using namespace std;
typedef pair<int,int> pi;
typedef pair<ll,ll> pl;

int main(){
    ll n;
    scanf(" %lld",&n);
    int ans = 0;
    ll upTo = sqrt(n*2LL);
    for(ll i = 1; i <= upTo;++i){
        if((n*2LL) % i == 0){
            ll see = 2LL*n - (i*i) + i;
            if(see % (2LL*i) == 0) ++ans;
            ll ot = (n*2LL) / i;
            if(ot != i && (2LL*n) % ot == 0){
                see = 2LL*n - (ot*ot) + ot;
                if(see % (2LL*ot) == 0) ++ans;
            }
        }
    }
    printf("%d\n",ans);
    return 0;
}
